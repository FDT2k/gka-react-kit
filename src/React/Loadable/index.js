import React from 'react';
import Loadable from 'react-loadable';
import Loading from '../Widgets/Loading'

export default  (loader)=>{
  const LoadableComponent =  Loadable({
    loader: loader,

    loading: (props)=>{
      if (props.error) {
        return <div>Error! {props.error.stack}<button onClick={ props.retry }>Retry</button></div>;
      } else if (props.pastDelay) {
        return (<Loading/>);
      } else {
        return null;
      }

    },
    delay: 500
  })


  return props=>(<LoadableComponent {...props}/>)
}
