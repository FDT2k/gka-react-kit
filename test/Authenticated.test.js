// Link.react.test.js
import "babel-polyfill"
import React from 'react';
import AuthenticatedHOC from '../src/React/HOC/AuthenticatedComponent';
import PublicHOC from '../src/React/HOC/PublicComponent';
import { Provider } from 'react-redux';
global.window = {}
import 'mock-local-storage'


import renderer from 'react-test-renderer';

import createStore,{createDefaultStoreWithMiddlewares} from '../src/Store';


import ReduxThunk from 'redux-thunk';

const cb_test_mw = cb => store => next => action => {
  cb(action);
  next(action);
}

let store = createDefaultStoreWithMiddlewares();

class TestComponent extends React.Component {


  render() {
    return (
      <h1>test</h1>
    );
  }
}

test('PublicHOC is rendering', () => {

  const C = PublicHOC(TestComponent)
  const component = renderer.create(
    <Provider store={store}>
      <C redirectTo="bla" history={[]}/>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});


test('PublicHOC is dipatching', () => {
  return new Promise ((resolve,reject)=>{

    let store = createStore({auth:{authenticated:true}},[ReduxThunk,cb_test_mw((action)=>{
      if(action.type == 'app/auth/UNAUTHENTICATED'){
        resolve();
      }
    })]);
    const C = PublicHOC(TestComponent)
    const component = renderer.create(
      <Provider store={store}>
        <C dispatch={true}/>
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});


test('AuthenticatedHOC is rendering', () => {

  const C = AuthenticatedHOC(TestComponent)
  const component = renderer.create(
    <Provider store={store}>
      <C redirectTo="bla" history={{push:()=>{
          console.log('ahi')
        }}}/>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});


test('AuthenticatedHOC is dispatching', () => {
  return new Promise ((resolve,reject)=>{
    const C = AuthenticatedHOC(TestComponent)
    let store = createStore({},[ReduxThunk,cb_test_mw((action)=>{
      if(action.type == 'app/auth/UNAUTHENTICATED'){
        resolve();
      }
    })]);

    const component = renderer.create(
      <Provider store={store}>
        <C redirectTo="bla" dispatch={true}
        />
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })


});
