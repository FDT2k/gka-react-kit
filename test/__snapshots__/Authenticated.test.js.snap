// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticatedHOC is dispatching 1`] = `
<h1>
  test
</h1>
`;

exports[`AuthenticatedHOC is rendering 1`] = `
<h1>
  test
</h1>
`;

exports[`PublicHOC is dipatching 1`] = `
<h1>
  test
</h1>
`;

exports[`PublicHOC is rendering 1`] = `
<h1>
  test
</h1>
`;
